File: farandula-java/src/main/java/com/nearsoft/farandula/models/AirLeg.java
package com.nearsoft.farandula.models;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Created by pruiz on 4/12/17.
 */
public class AirLeg {

    String id;
    String departureAirportCode;
    LocalDateTime departingDate;
    String arrivalAirportCode;
    LocalDateTime arrivalDate;
    List<Segment> segments;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDepartureAirportCode() {
        return departureAirportCode;
    }

    public void setDepartureAirportCode(String departureAirportCode) {
        this.departureAirportCode = departureAirportCode;
    }
    public LocalDateTime getDepartingDate() {
        return departingDate;
    }

    public void setDepartingDate(LocalDateTime departingDate) {
        this.departingDate = departingDate;
    }

    public String getArrivalAirportCode() {
        return arrivalAirportCode;
    }

    public void setArrivalAirportCode(String arrivalAirportCode) {
        this.arrivalAirportCode = arrivalAirportCode;
    }

    public LocalDateTime getArrivalDate() {
        return arrivalDate;
    }

    public void setArrivalDate(LocalDateTime arrivalDate) {
        this.arrivalDate = arrivalDate;
    }

    public List<Segment> getSegments() {
        return segments;
    }

    public void setSegments(List<Segment> segments) {
        this.segments = segments;
    }

}

File: farandula-js/src/models/Airleg.ts
class Airleg {
  private id:number
  private departureAirportCode:string
  private arrivalAirportCode:string
  private departingDate:number
  private arrivalDate:number
  private segments:Segment[]

  public getId(): number {
    return this.id
  }

  public setId(id:number): void {
    this.id = id
  }

  public getDepartureAirportCode(): string {
    return this.departureAirportCode
  }

  public setDepartureAirportCode(departureAirportCode:string): void {
    this.departureAirportCode = departureAirportCode
  }

  public getDepartingDate(): number {
    return this.departingDate
  }

  public setDepartingDate(departingDate:number): void {
    this.departingDate = departingDate
  }

  public getArrivalDate(): number {
    return this.arrivalDate
  }

  public setArrivalDate(arrivalDate:number): void {
    this.arrivalDate = arrivalDate
  }

  public getArrivalAirportCode(): string {
    return this.arrivalAirportCode
  }
  
  public setArrivalAirportCode(arrivalAirportCode:string): void {
    this.arrivalAirportCode = arrivalAirportCode
  }

}